src/main/java/org/zaproxy/zap/extension/hud/HudAPI.java

```diff
@@ -185,22 +185,22 @@ public boolean allowUnsafeEval() {
     public String handleCallBack(HttpMessage msg) throws ApiException {
         // Just used to handle files which need to be on the target domain
         try {
-            String query = msg.getRequestHeader().getURI().getPathQuery();
-            logger.debug("callback query = " + query);
-            if (query != null) {
-                if (query.indexOf("zapfile=") > 0) {
-                    String fileName = query.substring(query.indexOf("zapfile=") + 8);
-                    if (DOMAIN_FILE_WHITELIST.contains(fileName)) {
-                        msg.setResponseBody(
-                                this.getFile(msg, ExtensionHUD.TARGET_DIRECTORY + "/" + fileName));
-                        // Currently only javascript files are returned
-                        msg.setResponseHeader(
-                                API.getDefaultResponseHeader(
-                                        "application/javascript; charset=UTF-8",
-                                        msg.getResponseBody().length(),
-                                        false));
-                        return msg.getResponseBody().toString();
-                    }
+            String path = msg.getRequestHeader().getURI().getEscapedPath();
+            int lastSlash = path.lastIndexOf("/");
+            String fileName = path.substring(lastSlash + 1);
+
+            logger.debug("callback fileName = " + fileName);
+            if (fileName != null) {
+                if (DOMAIN_FILE_WHITELIST.contains(fileName)) {
+                    msg.setResponseBody(
+                            this.getFile(msg, ExtensionHUD.TARGET_DIRECTORY + "/" + fileName));
+                    // Currently only javascript files are returned
+                    msg.setResponseHeader(
+                            API.getDefaultResponseHeader(
+                                    "application/javascript; charset=UTF-8",
```

Comment:
Should this have a little more evaluation on file type for setting the header since we'll be serving CSS and image files as well?

Reply:
Those will be served by `HudFileProxy` which does that.